### PR TITLE
Basic 8.1.X support

### PR DESCRIPTION
**Pull request by funjoker**
Opcodes might not be 100%

Issues:

SMSG_QUERY_CREATURE_RESPONSE -> UnkFloat
SMSG_SPELL_START + SMSG_SPELL_START -> unkCount + unkByte
ContentTuning-> UnkUInt16
SMSG_PLAY_SPELL_VISUAL -> UnkUInt16
SMSG_ENUM_CHARACTERS_RESULT -> UnkCount+ 2*UInt32

Updatefields missing totaly.

### CODE REVIEW

**Comment by funjoker** [comment]
Travis is somewhat unable to get the Lib for DBC

**Comment by genion18** [comment]
Hy,

quest_request_items_locale/(36428, 'deDE', 'le HeilungIrgendwelche', 28768),
quest_offer_reward/(36428, 4, 378, 0, 0, 0, 0, 0, 0, 'le HeilungJa! Gute Arbeit, $n!$b$bIch habe schon mit einigen der Überlebenden ges', 28768), -- 36428
quest_greeting/(80157, 0, 0, 0, '\`Das Erbe der Apex', 28768), -- 80157
quest_request_items/(36428, 0, UNKNOWN, 0, UNKNOWN, 'le HeilungIrgendwelche', 28768), -- 36428

are Broken jet!

greets

**Comment by genion18** [comment]
Hy,

quest_request_items_locale/(36428, 'deDE', 'le HeilungIrgendwelche', 28768),
quest_offer_reward/(36428, 4, 378, 0, 0, 0, 0, 0, 0, 'le HeilungJa! Gute Arbeit, $n!$b$bIch habe schon mit einigen der Überlebenden ges', 28768), -- 36428
quest_greeting/(80157, 0, 0, 0, '��`Das Erbe der Apex', 28768), -- 80157
quest_request_items/(36428, 0, UNKNOWN, 0, UNKNOWN, 'le HeilungIrgendwelche', 28768), -- 36428
trainer/(373, 2, 'H', 28768);

are Broken jet!

greets

**Comment by funjoker** [comment]
It's basics only.

**Comment by genion18** [comment]
thanks ☺

**Comment by funjoker** [comment]
@genion18 if you find some parsing issues. just tell me. But not SMSG_UPDATE_OBJECT please ^^. It's totaly bugged atm

**Comment by genion18** [comment]
@funjoker Okay, thanks. I tell you when i find some Broken parsing issues. ☺
@Funjoker okay, danke. ich schreib dir wenn ich was finde.  ☺

File: WowPacketParser/Enums/Version/UpdateFields.cs [hunk 1]
```diff
@@ -88,7 +91,38 @@ static UpdateFields()
                         .Select(attribute => ((UpdateFieldAttribute)attribute).UFAttribute)
                         .DefaultIfEmpty(UpdateFieldType.Default).First();
 
-                    result.Add((int)vValues.GetValue(i), new UpdateFieldInfo() { Value = (int)vValues.GetValue(i), Name = vNames[i], Size = 0, Format = format });
+                    var vFormat = vEnumType.GetMember(vNames[i])
```
Comment (Warpten) [inline WowPacketParser/Enums/Version/UpdateFields.cs]: This is unreadable and duplicates code, make it a simple loop

Comment (funjoker) [reply WowPacketParser/Enums/Version/UpdateFields.cs]: well it actually loads attributes from general updatefieldslist
with this we can add attribute to the versioned updatefields

Comment (Warpten) [reply WowPacketParser/Enums/Version/UpdateFields.cs]: Yeah and it loads each property one by one, allocating a bunch of filter delegates. It makes it harder to maintain should we need to add properties, introduces code duplication, and overhead.

**Comment by xXBentRealityXx** [comment]
@funjoker Thanks man! Couldn't work on my repack properly without you ^^

**Comment by funjoker** [comment]
UFs is very experimental. Prolly won't be merged.
When someone made a good WPP implementation i will remove all UFs related commits and keep just the clean packet support.

**Comment by osleyder85** [comment]
funjoker Hello,
I just tried the new fixed fixes SMSG_QUEST_GIVER_QUEST_COMPLETE and they seemed very good to me. Thank you. Here I leave you some other things that I think are wrong:

INSERT INTO `gameobject_template_addon` (`entry`, `faction`, `flags`) VALUES 
(293684, 156698049, 369164288); [Err] 1264 - Out of range value for column 'faction' at row 1

Spawn creature: Do not appear the creatures id=0
(@CGUID+0, 0, 1, 0, 0, '0', '0', 0, 0, 0, -711.5295, -4217.467, 39.35896, 5.824013, 120, 0, 0, 0, 0, 0, 0, 0, 0, 29869), -- Northwatch Scout (Area: -Unknown- - Difficulty: 1) (Auras: 58506 - Stealth)

Spawn creature: The id=2818572288 of the creatures is incorrect.
(@CGUID+15, 2818572288, 1, 0, 0, '0', '0', 0, 0, 0, -569.441, -4264.957, 38.35974, 1.911952, 120, 0, 0, 0, 0, 0, 0, 0, 0, 29869), -- Foreman Thazz'ril (Area: -Unknown- - Difficulty: 1)

All data in zero.
INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `aiAnimKit`, `movementAnimKit`, `meleeAnimKit`, `auras`) VALUES
(@CGUID+0, 0, 0, 0, 0, 0, 0, 0, 0, '58506'), -- Northwatch Scout - 58506 - Stealth

Spawn gameobject: Do not appear the gameobject id=0
(@OGUID+3, 0, 1, 0, 0, '0', '', 0, -408.2125, -4395.518, 42.76613, 6.161013, 0, 0, -0.06104851, 0.9981348, 120, 0, 32, 29869), -- Cactus Apple (Area: -Unknown- - Difficulty: 1)

Spawn gameobject: The id=2818572288 of the gameobject is incorrect.
(@OGUID+4, 2818572288, 1, 0, 0, '0', '', 0, 0, 0, 0, 3.104126, 0, 0, 0.9998245, 0.01873213, 120, 124, 158, 29869), -- 0 (Area: -Unknown- - Difficulty: 1)

parent_rotation0=-34359740000 is incorrect.
INSERT INTO `gameobject_addon` (`guid`, `parent_rotation0`, `parent_rotation1`, `parent_rotation2`, `parent_rotation3`) VALUES
(@OGUID+0, -34359740000, 0, 0, 0), -- 0

All data in zero.
INSERT INTO `creature_template_addon` (`entry`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `aiAnimKit`, `movementAnimKit`, `meleeAnimKit`, `auras`) VALUES
(2630, 0, 0, 0, 0, 0, 0, 0, 0, '6474'), -- 2630 (Earthbind Totem) - Earthbind Totem Passive

creature_template: minlevel, maxlevel, faction is incorrect.
INSERT INTO `creature_template` (`entry`, `gossip_menu_id`, `minlevel`, `maxlevel`, `faction`, `npcflag`, `speed_walk`, `speed_run`, `BaseAttackTime`, `RangeAttackTime`, `unit_flags`, `unit_flags2`, `unit_flags3`, `dynamicflags`, `VehicleId`, `HoverHeight`) VALUES
(143509, 0, 1, 1, 35, 0, 1, 1, 2000, 2000, 0, 0, 0, 0, 0, 1), -- Training Dummy
(141968, 0, 1, 1, 35, 0, 1, 1.142857, 2000, 2000, 0, 0, 0, 1184, 0, 1), -- Wind Rider
(141963, 0, 1, 1, 35, 0, 1, 1.142857, 2000, 2000, 0, 0, 0, 1184, 0, 1), -- Darkspear Headhunter

I hope it helps.

**Comment by mdX7** [comment]
kinda everything db related is broken for 8.1.x atm since SMSG_UPDATE_OBJECT is not implemented properly yet. There was a big change in patch 8.1.0.

**Comment by osleyder85** [comment]
OK no problem. 
I just want to thank the work they are doing and thank them. I will try to help them as much as I can.

Greetings.

**Comment by funjoker** [comment]
Thanks. Infos are welcome. :D since I can’t test everything on my own, I do appreciate that info

File: WowPacketParser.sln [hunk 2]
```diff
@@ -48,68 +48,198 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WowPacketParserModule.V2_4_
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WowPacketParserModule.V8_0_1_27101", "WowPacketParserModule.V8_0_1_27101\WowPacketParserModule.V8_0_1_27101.csproj", "{749EA02D-C727-4697-B979-9ED80FB98085}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WowPacketParserModule.Substructures", "WowPacketParserModule.Substructures\WowPacketParserModule.Substructures.csproj", "{953229B7-01C4-457B-86CC-F0DD1173F217}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
```
Comment (Shauren) [inline WowPacketParser.sln]: Fix project and solution files to remove unneccessary x64 and x86 builds, you should only have "Any CPU"

File: WowPacketParserModule.V8_0_1_27101/Parsers/SpellHandler.cs [hunk 3]
```diff
@@ -143,6 +217,8 @@ public static void ReadContentTuningParams(Packet packet, params object[] idx)
             packet.ReadInt16("PlayerLevelDelta", idx);
             packet.ReadUInt16("PlayerItemLevel", idx);
             packet.ReadUInt16("ScalingHealthItemLevelCurveID", idx);
+            if (ClientVersion.AddedInVersion(ClientVersionBuild.V8_1_0_28724))
+                packet.ReadUInt16("UnkUInt16_810", idx);
```
Comment (Shauren) [inline WowPacketParserModule.V8_0_1_27101/Parsers/SpellHandler.cs]: TargetItemLevel, between PlayerItemLevel and CurveID (used for pvp)

File: WowPacketParserModule.V8_0_1_27101/Parsers/AuthenticationHandler.cs [hunk 4]
```diff
@@ -78,8 +78,14 @@ public static void HandleAuthResponse(Packet packet)
                     packet.ResetBitReader();
                     packet.ReadBit("IsLocal", "VirtualRealms", i);
                     packet.ReadBit("IsInternalRealm", "VirtualRealms", i);
-                    var nameLen1 = packet.ReadBits(8);
-                    var nameLen2 = packet.ReadBits(8);
+
+                    var bitsCount = 8;
+
+                    if (ClientVersion.AddedInVersion(ClientVersionBuild.V8_1_0_28724))
+                        bitsCount = 9;
```
Comment (Shauren) [inline WowPacketParserModule.V8_0_1_27101/Parsers/AuthenticationHandler.cs]: not seeing this in 8.1.5, still 8 bits

**Pull request merged**